### PR TITLE
Upgrade Outscale provider

### PR DESCRIPTION
**Pull request by outscale-mdr**
- Upgrade terraform outscale provider
- Fix iops default value when using other disk types than io1
- Fix public cloud deployment (Ip conflict)

### CODE REVIEW

File: providers.tf [hunk 1]
```diff
@@ -1,8 +1,7 @@
 terraform {
   required_providers {
     outscale = {
-      source  = "outscale-dev/outscale"
-      version = "0.5.0"
+      source  = "outscale/outscale"
```
Comment (outscale-vbr) [inline providers.tf:4]: Missing version ?

File: control-planes.tf [hunk 2]
```diff
@@ -91,7 +91,7 @@ resource "outscale_vm" "control-planes" {
       delete_on_vm_deletion = true
       volume_size           = var.control_plane_volume_size
       volume_type           = var.control_plane_volume_type
-      iops                  = var.control_plane_volume_type == "io1" ? var.control_plane_iops : 0
+      iops                  = var.control_plane_volume_type == "io1" ? var.control_plane_iops : null
```
Comment (outscale-vbr) [inline control-planes.tf:94]: Type iops change ?
https://registry.terraform.io/providers/outscale/outscale/latest/docs/resources/vm
If yes, there is no changes in docs.

File: workers.tf [hunk 3]
```diff
@@ -76,7 +76,7 @@ resource "outscale_vm" "workers" {
     connection {
       type                = "ssh"
       user                = "outscale"
-      host                = var.public_cloud ? outscale_public_ip.workers[count.index].public_ip : format("10.0.1.%d", 19 + count.index)
+      host                = var.public_cloud ? self.public_ip : format("10.0.1.%d", 19 + count.index)
```
Comment (outscale-vbr) [inline workers.tf:79]: You use this self ?
https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax

File: providers.tf (hunk 1 above)
Comment (outscale-mdr) [reply providers.tf:4]: Always the latest, no ?

File: control-planes.tf (hunk 2 above)
Comment (outscale-vbr) [reply control-planes.tf:94]: Ok good

**Review by outscale-vbr — APPROVED** [review]

File: providers.tf (hunk 1 above)
Comment (outscale-vbr) [reply providers.tf:4]: Ok

File: control-planes.tf (hunk 2 above)
Comment (outscale-vbr) [reply control-planes.tf:94]: (0  is incompatible with gp2)

**Pull request merged**